src/sentry/snuba/metrics/helpers.py

```diff
@@ -0,0 +1,694 @@
+
+    date_range = end - start
+
+    date_range = timedelta(seconds=int(interval * math.ceil(date_range.total_seconds() / interval)))
+
+    if date_range.total_seconds() / interval > MAX_POINTS:
+        raise InvalidParams(
+            "Your interval and date range would create too many results. "
+            "Use a larger interval, or a smaller date range."
+        )
+
+    end_ts = int(interval * math.ceil(to_timestamp(end) / interval))
+    end = to_datetime(end_ts)
+    start = end - date_range
+
+    # NOTE: The sessions_v2 implementation cuts the `end` time to now + 1 minute
+    # if `end` is in the future. This allows for better real time results when
+    # caching is enabled on the snuba queries. Removed here for simplicity,
+    # but we might want to reconsider once caching becomes an issue for metrics.
+
+    return start, end, interval
+
+
+#: The type of metric, which determines the snuba entity to query
+MetricType = Literal["counter", "set", "distribution"]
+
+#: A function that can be applied to a metric
+MetricOperation = Literal["avg", "count", "max", "min", "p50", "p75", "p90", "p95", "p99"]
+
+MetricUnit = Literal["seconds"]
+
+
+METRIC_TYPE_TO_ENTITY: Mapping[MetricType, EntityKey] = {
+    "counter": EntityKey.MetricsCounters,
+    "set": EntityKey.MetricsSets,
+    "distribution": EntityKey.MetricsDistributions,
+}
+
+
+class MetricMeta(TypedDict):
+    name: str
+    type: MetricType
+    operations: Collection[MetricOperation]
+    unit: Optional[MetricUnit]
+
+
+class Tag(TypedDict):
+    key: str  # Called key here to be consistent with JS type
+
+
+class TagValue(TypedDict):
+    key: str
+    value: str
+
+
+class MetricMetaWithTagKeys(MetricMeta):
+    tags: Sequence[Tag]
+
+
+# Map requested op name to the corresponding Snuba function
+_OP_TO_SNUBA_FUNCTION = {
+    "metrics_counters": {"sum": "sum"},
+    "metrics_distributions": {
+        "avg": "avg",
+        "count": "count",
+        "max": "max",
+        "min": "min",
+        # TODO: Would be nice to use `quantile(0.50)` (singular) here, but snuba responds with an error
+        "p50": "quantiles(0.50)",
+        "p75": "quantiles(0.75)",
+        "p90": "quantiles(0.90)",
+        "p95": "quantiles(0.95)",
+        "p99": "quantiles(0.99)",
+    },
+    "metrics_sets": {"count_unique": "uniq"},
+}
+
+AVAILABLE_OPERATIONS = {
+    type_: sorted(mapping.keys()) for type_, mapping in _OP_TO_SNUBA_FUNCTION.items()
+}
+
+
+_BASE_TAGS = {
+    "environment": [
+        "production",
+        "staging",
+    ],
+    "release": [],
+}
+
+_SESSION_TAGS = dict(
+    _BASE_TAGS,
+    **{
+        "session.status": [
+            "abnormal",
+            "crashed",
+            "errored",
+            "healthy",
+        ],
+    },
+)
+
+_TRANSACTION_TAGS = dict(
+    _BASE_TAGS,
+    transaction=["/foo/:orgId/", "/bar/:orgId/"],
+)
+
+_MEASUREMENT_TAGS = dict(
+    _TRANSACTION_TAGS,
+    measurement_rating=["good", "meh", "poor"],
+)
+
+METRICS = {
```

Comment:
I would actually like to get rid of this entire dict, but let's do that in a follow up PR. I will create a Jira issue for that.

Reply:
Oh wait, is this now only used in `_get_metric`? In that case I would remove it right away. We don't need to return a 400 response if the metric does not exist in snuba, an empty result is fine.

Reply:
Yeah it seems that's the only place its used for, I agree with your suggestions.. so how would you edit `_get_metric` ?

Reply:
I would remove it entirely and use the function name (`op`) to map to the correct metrics type here instead:
https://github.com/getsentry/sentry/blob/ddb6ef15e2036ac4c9953a2266224b2fe983d992/src/sentry/snuba/metrics/helpers.py#L532-L534

If you think this is out of scope for the PR, I'm happy to do it in a follow-up.

Reply:
I would remove it entirely and use the function name (`op`) to map to the correct metrics type here instead:
https://github.com/getsentry/sentry/blob/ddb6ef15e2036ac4c9953a2266224b2fe983d992/src/sentry/snuba/metrics/helpers.py#L532-L534

If you think this is out of scope for the PR, I'm happy to do it in a follow-up.

Reply:
Will do a follow up PR like we discussed on slack